### PR TITLE
Removing @symbolic from few tf backend ops, Adding `Metric`, `Sum`, `Mean` classes.

### PR DESCRIPTION
**Pull request by pavithrasv**
### Summary
1. Removing @symbolic from few tf backend ops and removing control dependencies from the update ops.
2. Adding `Metric` class.
3. Adding `Sum` metric class.
4. Adding `Mean` metric class.
5. Moving metrics_test.py to metrics_functional_test.py

### Related Issues
n/a

### PR Overview

- [y] This PR requires new unit tests [y/n] (make sure tests are included)
- [n ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y ] This PR is backwards compatible [y/n]
- [n ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/metrics.py [hunk 1]
```diff
@@ -19,10 +23,241 @@
...
+        self.var.assign_add(...).
+    * `result()`: Computes and returns a value for the metric
+        from the state variables.
+    """
+
+    def __init__(self, name=None, dtype=None, **kwargs):
+        super(Metric, self).__init__(name=name, dtype=dtype, **kwargs)
+        self.stateful = True  # All metric layers are stateful.
+        self.built = True
+        self.dtype = K.floatx() if dtype is None else dtype
```
Comment (fchollet) [inline keras/metrics.py]: `self.dtype = dtype or K.floatx()` is equivalent and simpler.

File: tests/keras/metrics_test.py [hunk 2]
```diff
@@ -1,238 +1,173 @@
...
-                               atol=1e-5)
-    np.testing.assert_allclose(val_outs[2],
-                               history.history['val_true_positives'][-1],
-                               atol=1e-5)
-
-
-if __name__ == '__main__':
-    pytest.main([__file__])
+
+class TestSum:
```
Comment (fchollet) [inline tests/keras/metrics_test.py]: Classes should inherit `object`

**Comment by pavithrasv** [comment]
> Thank you for the PR.
> 
> The following tests are currently failing _when running with TF1_. Note we must support both TF1 and TF2 with a single release, since there can only be one release of Keras as default on PyPI at any given time. It's fine to have TF1-specific or TF2-specific statements (abstracted in the backend).
> 
> ```
> tests/keras/metrics_test.py::TestSum::test_sum
> tests/keras/metrics_test.py::TestMean::test_mean
> tests/keras/metrics_test.py::TestSum::test_sum_with_sample_weight
> tests/keras/metrics_test.py::TestMean::test_mean_with_sample_weight
> tests/keras/metrics_test.py::TestMean::test_mean
> tests/keras/metrics_test.py::TestMean::test_multiple_instances
> tests/keras/metrics_test.py::TestMean::test_mean_with_sample_weight
> tests/keras/metrics_test.py::TestMean::test_multiple_instances
> ```
> 
> The failures seem to be caused by update ops not running in the `__call__` use case.
> 
> I'm not entirely sure what's the best path forward. We should either introduce a notion of "control dependency" in the backend (best UX), or perhaps remove `__call__` from the API since it cannot be used correctly.

I have added control_dependencies op and kept the __call__ API as it will be required in future for some metric implementations anyway.

**Comment by pavithrasv** [comment]
> > Thank you for the PR.
> > The following tests are currently failing _when running with TF1_. Note we must support both TF1 and TF2 with a single release, since there can only be one release of Keras as default on PyPI at any given time. It's fine to have TF1-specific or TF2-specific statements (abstracted in the backend).
> > ```
> > tests/keras/metrics_test.py::TestSum::test_sum
> > tests/keras/metrics_test.py::TestMean::test_mean
> > tests/keras/metrics_test.py::TestSum::test_sum_with_sample_weight
> > tests/keras/metrics_test.py::TestMean::test_mean_with_sample_weight
> > tests/keras/metrics_test.py::TestMean::test_mean
> > tests/keras/metrics_test.py::TestMean::test_multiple_instances
> > tests/keras/metrics_test.py::TestMean::test_mean_with_sample_weight
> > tests/keras/metrics_test.py::TestMean::test_multiple_instances
> > ```
> > 
> > 
> > The failures seem to be caused by update ops not running in the `__call__` use case.
> > I'm not entirely sure what's the best path forward. We should either introduce a notion of "control dependency" in the backend (best UX), or perhaps remove `__call__` from the API since it cannot be used correctly.
> 
> I have added control_dependencies op and kept the **call** API as it will be required in future for some metric implementations anyway.

Added no-op control dependency ops to Theano and CNTK backends. Fixed test failures in TF 1.X. Please take a look.

**Pull request merged**